apps/checker/app/matchers/DictionaryMatcher.scala

```diff
@@ -33,18 +33,35 @@ class DictionaryMatcher(
 
   val matcher = new LanguageToolMatcher(instance)
 
+  def matchFallsWithinNamedEntityRange(
+      ruleMatch: RuleMatch,
+      entities: List[EntityInText]
+  ): Boolean = {
+    entities.exists(entity =>
+      entity.range.from == ruleMatch.fromPos && entity.range.to == ruleMatch.toPos
+    )
+  }
+
   override def check(
       request: MatcherRequest
-  )(implicit ec: ExecutionContext) = {
-    // groupKey is used to control how rules are grouped in the client when they produces matches.
-    // This is needed for dictionary matches as they all share a common rule ID (MORFOLOGIK_RULE_COLLINS)
-    // groupKeys for dictionary matches have the format `MORFOLOGIK_RULE_COLLINS-{matchedText}`
+  )(implicit ec: ExecutionContext): Future[List[RuleMatch]] = {
+    val entityHelper = new EntityHelper()
+    val namedEntities =
+      request.blocks.flatMap((block) => entityHelper.getEntitiesFromText(block.text))
```

Comment:
We'll need to include the offset of the current block when marking entity ranges – entity ranges will begin at the start of their block, but match ranges will be relative to the entire request, so if the block doesn't start at 0, entities and ranges won't match up!

I can see this when checking on CODE – every prosemirror document begins at position `2`, so entities and matches don't line up. (Also should be verifiable with a test for a block that starts at a nonzero position.)

This should be as simple as 

```
TextRange(
  from = from + block.from,
  to = to + block.from
)
```

Passing an `offset` into `getEntitiesFromText` would be one way of solving.

Reply:
Good catch, thank you. I've attempted a fix here: 

https://github.com/guardian/typerighter/pull/443/commits/765e886db51e4048ab869d38077765f859bd999d

Will test on CODE shortly to verify